### PR TITLE
[Telemetry][Misc] Client-Side changes to log run results at the end of runs

### PR DESCRIPTION
**Pull request by frutescens**
## What are the changes the user will see?
None.

## Why am I making these changes?
Telemetry. 

## What are the changes from a developer perspective?
The /newclear request sent in game-over-phase now also sends the result and the game mode as part of the URL. 
See PR [4 ](https://github.com/f-fsantos/rogueserver/pull/4) for server side code. 

## How to test the changes?
Finish a run through winning or losing. 

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/@types/PokerogueSessionSavedataApi.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export class UpdateSessionSavedataRequest {
 export interface NewClearSessionSavedataRequest {
   slot: number;
   clientSessionId: string;
+  result: boolean;
```
Comment (Tempo-anon) [inline src/@types/PokerogueSessionSavedataApi.ts:12]: this name is not really clear to me what it means. As a boolean, the naming should be something like `isX` or `hasX`. Judging by the game-over-phase, this should be something like isVictory

**Review by Tempo-anon — COMMENTED** [review]
Would be curious to know what % of players just save over a run they don't want to continue

File: src/@types/PokerogueSessionSavedataApi.ts (hunk 1 above)
Comment (frutescens) [reply src/@types/PokerogueSessionSavedataApi.ts:12]: image
the result looks like this... which is why its called that

Comment (frutescens) [reply src/@types/PokerogueSessionSavedataApi.ts:12]: this is separate from players overwriting a run btw

Comment (f-fsantos) [reply src/@types/PokerogueSessionSavedataApi.ts:11]: the metric name doesn't have to force the system, we can name it something in the endpoint and something else in prometheus.

File: src/phases/game-over-phase.ts [hunk 2]
```diff
@@ -26,36 +26,36 @@ import i18next from "i18next";
 import { pokerogueApi } from "#app/plugins/api/pokerogue-api";
 
 export class GameOverPhase extends BattlePhase {
-  private victory: boolean;
+  private isVictory: boolean;
   private firstRibbons: PokemonSpecies[] = [];
 
   constructor(scene: BattleScene, victory?: boolean) {
```
Comment (DayKev) [inline src/phases/game-over-phase.ts:32]:
```suggestion
  constructor(scene: BattleScene, victory: boolean = false) {
```

File: src/phases/game-over-phase.ts [hunk 3]
```diff
@@ -26,36 +26,36 @@ import i18next from "i18next";
...
     super(scene);
 
-    this.victory = !!victory;
+    this.isVictory = !!victory;
```
Comment (DayKev) [inline src/phases/game-over-phase.ts:35]:
```suggestion
    this.isVictory = victory;
```

**Review by DayKev — APPROVED** [review]

File: src/phases/game-over-phase.ts [hunk 4]
```diff
@@ -176,22 +176,18 @@ export class GameOverPhase extends BattlePhase {
     /* Added a local check to see if the game is running offline on victory
       If Online, execute apiFetch as intended
       If Offline, execute offlineNewClear(), a localStorage implementation of newClear daily run checks */
-    if (this.victory) {
```
Comment (torranx) [inline src/phases/game-over-phase.ts:179]: is it intended for this condition to be removed? 
docs above says `Added a local check to see if the game is running offline on victory`

Comment (frutescens) [reply src/phases/game-over-phase.ts:179]: You're right! I fixed it!

**Review by DayKev — APPROVED** [review]

File: src/phases/game-over-phase.ts [hunk 5]
```diff
@@ -173,25 +173,23 @@ export class GameOverPhase extends BattlePhase {
...
-      if (!Utils.isLocal || Utils.isLocalServerConnected) {
-        pokerogueApi.savedata.session.newclear({ slot: this.scene.sessionSlotId, clientSessionId })
-          .then((success) => doGameOver(!!success));
-      } else {
+      If Offline, execute offlineNewClear() only for victory, a localStorage implementation of newClear daily run checks */
+    if (!Utils.isLocal || Utils.isLocalServerConnected) {
+      pokerogueApi.savedata.session.newclear({ slot: this.scene.sessionSlotId, isVictory: this.isVictory, clientSessionId: clientSessionId })
+        .then((success) => doGameOver(!!success));
+    } else {
+      if (this.isVictory) {
```
Comment (torranx) [inline src/phases/game-over-phase.ts:183]: nit: the `if` block can be merged into `else if`

File: src/phases/game-over-phase.ts [hunk 6]
```diff
@@ -26,36 +26,36 @@ import i18next from "i18next";
 import { pokerogueApi } from "#app/plugins/api/pokerogue-api";
 
 export class GameOverPhase extends BattlePhase {
-  private victory: boolean;
+  private isVictory: boolean;
   private firstRibbons: PokemonSpecies[] = [];
 
-  constructor(scene: BattleScene, victory?: boolean) {
+  constructor(scene: BattleScene, victory: boolean = false) {
```
Comment (torranx) [inline src/phases/game-over-phase.ts:32]: nit: (might as well change this too
```suggestion
  constructor(scene: BattleScene, isVictory: boolean = false) {
```

**Review by torranx — APPROVED** [review]

**Pull request merged**